routes.js

```diff
@@ -111,7 +111,12 @@ module.exports = async function (app, opts) {
       }
     }
 
-    return reply.graphql(query, null, variables, operationName)
+    let context
+    if (opts.context) {
+      context = await opts.context(request, reply)
+    }
```

Comment:
Would you mind extracting `opts.context` into a `const` at the top of the plugin?